crates/proto/src/op/message.rs

```diff
@@ -780,10 +797,10 @@ where
     let mut additional_count = count_was_truncated(additionals.emit(encoder))?;
 
     if let Some(edns) = edns {
+        encoder.set_max_size(real_encoder_max_size);
         // need to commit the error code
```

Comment:
What does this comment mean btw?

Reply:
I think that is referring to the fact that the response error code may need to be committed to the EDNS record... i.e. in EDNS the response (error code) is has high-order bits stored in the EDNS record that augment those in the original DNS Header. So I think that's what this may be talking about...